tlatools/org.lamport.tlatools/test/util/ParserInterface.java

```diff
@@ -0,0 +1,108 @@
+package util;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import tla2sany.parser.SyntaxTreeNode;
+import tla2sany.parser.TLAplusParser;
+import tla2sany.semantic.AbortException;
+import tla2sany.semantic.Context;
+import tla2sany.semantic.Errors;
+import tla2sany.semantic.Generator;
+import tla2sany.semantic.ModuleNode;
+import tla2sany.semantic.SemanticNode;
+
+/**
+ * A simple interface for parsing self-contained TLA+ modules, for use in
+ * tests. This class can be viewed as a prototype of a future API to be
+ * integrated into SANY itself.
+ */
+public class ParserInterface {
```

Comment:
Terminology: "Interface" yet it is a Java class.

Reply:
I was thinking of calling it API. Do you want to use ParserAPI or ParserApi casing?